conftest.py

```diff
@@ -192,12 +193,24 @@ def setup_artifact_path(item: pytest.Item, artifact_name: str):
     return os.path.join(output_dir, artifact_name)
 
 
-def get_task_files_for_id(task_id: str) -> set:
+def get_task_files_for_id(task_id: str) -> dict:
     try:
-        return set(subprocess.check_output(['dcos', 'task', 'ls', task_id, '--all']).decode().split())
+        # Example output:
+        # drwxr-xr-x  6  nobody  nobody      4096  Jul 21 22:07                             jre1.8.0_144
+        # drwxr-xr-x  3  nobody  nobody      4096  Jun 28 12:50                          libmesos-bundle
+        # -rw-r--r--  1  nobody  nobody  32539549  Jan 04 16:31  libmesos-bundle-1.10-1.4-63e0814.tar.gz
+        ls_lines = subprocess.check_output(['dcos', 'task', 'ls', task_id, '--long', '--all']).decode().split('\n')
+        ret = {}
+        for line in ls_lines:
+            # example elems: ['drwxr-xr-x', '6', 'nobody', 'nobody', '4096', 'Jul 21 22:07', 'jre1.8.0_144']
+            elems = [s.strip() for s in filter(None, line.split('  '))]
```

Comment:
Does the filter not mean that the length of `elems` could be variable?

Reply:
In theory this is just to remove some empty entries produced by the `split()`
Actually, it does look like there's some problems with this current handling, note the preceding space on the resulting filename:

```
>>> s = 'drwxr-xr-x  6  nobody  nobody      4096  Jul 21 22:07                             jre1.8.0_144'
>>> s.split('  ')
['drwxr-xr-x', '6', 'nobody', 'nobody', '', '', '4096', 'Jul 21 22:07', '', '', '', '', '', '', '', '', '', '', '', '', '', ' jre1.8.0_144']
>>> [e for e in filter(None, s.split('  '))]
['drwxr-xr-x', '6', 'nobody', 'nobody', '4096', 'Jul 21 22:07', ' jre1.8.0_144']
```

Going to figure out how to do this properly real quick..

Reply:
Ok, replaced the whole thing with a regex. Should be much less brittle now.